packages/venia-ui/lib/components/App/app.js

```diff
@@ -33,6 +37,30 @@ const App = props => {
     const { markErrorHandled, renderError, unhandledErrors } = props;
 
     const [, { addToast }] = useToasts();
+    const [{ cartId }, { getCartDetails, setCartId }] = useCartContext();
+    const [, checkoutActions] = useCheckoutContext();
+    const [createCart] = useMutation(CREATE_CART_MUTATION);
+
+    // On initial mount create a cart if there isn't a cartId in the store.
+    // TODO: Should this belong elsewhere? Basically this is global logic that says "any time the cartId is deleted, recreate the cart/do other actions".
+    useEffect(() => {
+        async function initializeCart() {
+            // First, reset checkout to clear any old state.
+            checkoutActions.actions.reset();
+
+            // Then fetch the cartId (existing or new) and store it.
+            const {
+                data: { createEmptyCart }
+            } = await createCart();
+            await setCartId(createEmptyCart);
+
+            getCartDetails({ forceRefresh: true });
+        }
+
+        if (!cartId) {
+            initializeCart();
```

Comment:
Now, as a side effect of not having a cartId we re-create one.

Ideally we never call the createCart action as it uses REST. We want to use the createCart mutation instead.

In this PR you'll see places I removed calls that would create a new cart through REST. Instead, we now fetch a new cart, set the cartId, and then fetch the details for that cart.

Reply:
Consider making a custom hook to contain this effect, so as not to bloat `App` with cart code.

Reply:
Done!